### PR TITLE
Crash on oom

### PR DESCRIPTION
**Pull request by msantos**
Some small safety fixes, if anything looks weird or you'd like some changes, please let me know! I've run the tests and everything passed.

### CODE REVIEW

**Comment by jaredmorrow** [comment]
@msantos I wish I noticed this PR last week.   I fixed a few bugs in erlang_js that were causing segfaults on otp 15B and having never looked at the erlang_js code I didn't know we were so vulnerable to OOM.   Thanks for the PR, I'll try and look at this after our next release.

**Comment by msantos** [comment]
Thanks for looking at it! I've reapplied my patches on top of your changes.

File: c_src/spidermonkey.c [hunk 1]
```diff
@@ -240,8 +241,8 @@ char *escape_quotes(char *text) {
 char *error_to_json(const spidermonkey_error *error) {
   char *escaped_source = escape_quotes(error->offending_source);
   /* size = length(escaped source) + length(error msg) + JSON formatting */
-  size_t size = (strlen(escaped_source) + strlen(error->msg) + 80) * sizeof(char);
-  char *retval = (char *) driver_alloc((ErlDrvSizeT) size);
+  size_t size = strlen(escaped_source) + strlen(error->msg) + 80;
```
Comment (jaredmorrow) [inline c_src/spidermonkey.c:244]: `* sizeof(char);` doesn't seem like it should've been removed

Comment (msantos) [inline c_src/spidermonkey.c:244]: On Tue, Jun 12, 2012 at 09:03:24AM -0700, Jared Morrow wrote:

> > @@ -240,8 +241,8 @@ char _escape_quotes(char *text) {
> >  char *error_to_json(const spidermonkey_error *error) {
> >    char *escaped_source = escape_quotes(error->offending_source);
> >    /_ size = length(escaped source) + length(error msg) + JSON formatting */
> > -  size_t size = (strlen(escaped_source) + strlen(error->msg) + 80) \* sizeof(char);
> > -  char *retval = (char *) driver_alloc((ErlDrvSizeT) size);
> > -  size_t size = strlen(escaped_source) + strlen(error->msg) + 80;
> 
> `* sizeof(char);` doesn't seem like it should've been removed

sizeof(char) will always be 1. Removing it is sort of pedantic, but it
simplifies the code a bit.

Comment (jaredmorrow) [inline c_src/spidermonkey.c:244]: Please ignore me while I confuse sizeof(char) with CHAR_BITS, sorry.

**Comment by Skamatam** [comment]
+1

**Comment by joecaswell** [comment]
+1

**Comment by chardan** [comment]
+1 Looks good.

It might be nice for maintenance to peek at (driver_comm.c, 37) using the copy_string()
function here instead of the bare code if it's safe (it looks ilke we're still dealing with NULL-terminated strings).

Not created in this change, but copy_string() might be better-named ejs_strdup() to be
more consistent with the standard library in a future pass.

**Pull request merged**